cmd/account/rotate-secret.go

```
@@ -80,128 +57,171 @@ func newRotateSecretOptions(streams genericclioptions.IOStreams, flags *genericc
 }
 
 func (o *rotateSecretOptions) complete(cmd *cobra.Command, args []string) error {
+
 	if len(args) != 1 {
-		return cmdutil.UsageErrorf(cmd, "IAM User name argument is required")
+		return cmdutil.UsageErrorf(cmd, "Account CR argument is required")
 	}
-	o.iamUsername = args[0]
 
-	// account CR name and account ID cannot be empty at the same time
-	if o.accountName == "" && o.accountID == "" {
-		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be empty at the same time")
-	}
+	o.accountCRName = args[0]
 
-	if o.accountName != "" && o.accountID != "" {
-		return cmdutil.UsageErrorf(cmd, "AWS account CR name and AWS account ID cannot be set at the same time")
+	if o.profile == "" {
+		o.profile = "default"
```

what happens if default profile doesn't exist?

---

Then creating a client will throw an error. This is just following how the aws cli works.

---

I *doubt* this would bite anyone, but silently defaulting to the default account kinda creeps me out.  Should we not require the user to explicitly say which account they want to use?

---

I'm in the camp of silently defaulting to the default account set, just to align our user-experience with that of the aws cli, like Taylor mentioned.

---

+1 to following the same behaviour as aws cli regarding the default profile